emcache/cluster.py

```diff
@@ -221,6 +257,82 @@ def node(self, memcached_host_address: MemcachedHostAddress) -> Node:
 
         raise ValueError(f"{memcached_host_address} can not be found")
 
+    async def _autodiscovery_monitor(self) -> None:
+        logger.debug("Autodiscovery task started")
+        while True:
+            try:
+                if await self.autodiscover():
+                    await asyncio.sleep(self._autodiscovery_poll_interval)
+                else:
+                    await asyncio.sleep(AUTODISCOVERY_POLL_INTERVAL_FAILURE)
+            except asyncio.CancelledError:
+                # if a cancellation is received we stop monitoring for cluster changes
+                break
+
+            except Exception:
+                logger.exception("Autodiscover raised an exception, continuing processing")
+
+        logger.debug("Autodiscovery task stopped")
+
+    async def _get_autodiscovered_nodes(self) -> tuple[bool, int, list[tuple[str, str, int]]]:
+        if self._closed:
+            raise RuntimeError("Emcache client is closed")
+
+        node = random.choice(self._original_nodes)
+        async with OpTimeout(self._autodiscovery_timeout, self._loop):
+            async with node.connection() as connection:
+                return await connection.autodiscovery()
+
+    async def autodiscover(self) -> bool:
```

Comment:
Hey Im currently on vacations, Im quite limitted for making reviews, so be patience :)

In any case this function should have proper coverage,  unless it has already, for checking the different branches i.e new nodes added, nodes removed, ...

Reply:
I added unit tests and that fuction is now 100% covered.